pkg/manifests/manifests.go

```diff
@@ -2140,7 +2140,8 @@ func (f *Factory) setTLSSecurityConfiguration(args []string, tlsCipherSuitesArg
 func setArg(args []string, argName string, argValue string) []string {
 	found := false
 	for i, arg := range args {
-		if strings.HasPrefix(arg, argName) {
+		if arg == argName ||
+			(len(argName) > 0 && argName[len(argName)-1] == '=' && strings.HasPrefix(arg, argName)) {
```

Comment:
Is there a case where `argName` can be empty or is it just by precaution?

Reply:
this is by precaution, preventing a potential out of index error :)

Reply:
IIUC an empty `argName` can't be triggered by user input (e.g. via CMO configmap) so I think that it's better to fail. If we hit this situation, it means that we have introduced a bug in the CMO logic.

```suggestion
			(argName[len(argName)-1] == '=' && strings.HasPrefix(arg, argName)) {
```